router/batchrouter/batchrouter.go

```diff
@@ -422,15 +422,18 @@ func (brt *HandleT) pollAsyncStatus(ctx context.Context) {
 											statusList = append(statusList, status)
 										}
 										brt.failedJobCount.Count(len(statusList))
-										txn := brt.jobsDB.BeginGlobalTransaction()
-										brt.jobsDB.AcquireUpdateJobStatusLocks()
-										err = brt.jobsDB.UpdateJobStatusInTxn(txn, statusList, []string{brt.destType}, parameterFilters)
+										err := brt.jobsDB.WithUpdateSafeTx(func(tx jobsdb.UpdateSafeTx) error {
+											err = brt.jobsDB.UpdateJobStatusInTx(tx, statusList, []string{brt.destType}, parameterFilters)
```

Comment:
`UpdateJobStatusInTx` do we need to perform that unsafely? If not it makes sense to have the UpdateSafeTx as part of the `UpdateJobStatusInTx` of we should decorate the later to and make the decorated version public.

Reply:
can you provide an example?